### PR TITLE
fix support for order by _raw & _id

### PR DESCRIPTION
**Pull request by seut**

### CODE REVIEW

File: sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java [hunk 1]
```diff
@@ -115,8 +111,38 @@ public SortField visitReference(final Reference symbol, final SortSymbolContext
...
+
+        if (columnIdent.isColumn()) {
+            if (SortParseElement.SCORE_FIELD_NAME.equals(columnIdent.name())) {
+                return !context.reverseFlag ? SortParseElement.SORT_SCORE_REVERSE : SortParseElement.SORT_SCORE;
+            } else if (DocSysColumns.RAW.equals(columnIdent) || DocSysColumns.ID.equals(columnIdent)) {
+                String luceneName = DocSysColumns.nameForLucene(columnIdent);
+                CollectInputSymbolVisitor.Context inputContext = inputSymbolVisitor.extractImplementations(symbol);
+                ArrayList<Input<?>> inputs = inputContext.topLevelInputs();
+                assert inputs.size() == 1;
+                final Input functionInput = inputs.get(0);
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java]: why is this called functionInput?

File: sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java [hunk 2]
```diff
@@ -115,8 +111,38 @@ public SortField visitReference(final Reference symbol, final SortSymbolContext
...
+                @SuppressWarnings("unchecked")
+                final List<LuceneCollectorExpression> expressions = inputContext.docLevelExpressions();
+
+                return new SortField(luceneName, new IndexFieldData.XFieldComparatorSource() {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java]: would it make sense to create an inner static class for the anonymous XFieldComparatorSource? The one that is created for Function symbols looks pretty much the same

Comment (mfussenegger) [inline sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java]: actually most looks the same except the type/reducedType thing

File: sql/src/main/java/io/crate/operation/reference/doc/lucene/RawCollectorExpression.java [hunk 3]
```diff
@@ -41,6 +44,10 @@ public void startCollect(CollectorContext context) {
 
     @Override
     public BytesRef value() {
-        return visitor.source().toBytesRef();
+        try {
+            return CompressorFactory.uncompressIfNeeded(visitor.source()).toBytesRef();
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/reference/doc/lucene/RawCollectorExpression.java]: raw is currently in use by copy from / to, please check how/why it works there and if uncompressing it here would slow things down

File: sql/src/main/java/io/crate/action/sql/query/FieldsVisitorInputFieldComparator.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import java.io.IOException;
+import java.util.List;
+
+public class FieldsVisitorInputFieldComparator extends InputFieldComparator {
+
+    private final CollectorContext collectorContext;
+    private IndexReader currentReader;
+
+
+    public FieldsVisitorInputFieldComparator(int numHits,
```
Comment (dobe) [inline sql/src/main/java/io/crate/action/sql/query/FieldsVisitorInputFieldComparator.java]: do you need the collectorcontext here? would probably make sense to pass the visitor in the constructor

**Comment by dobe** [comment]
changes entry is missing

**Comment by dobe** [comment]
solve conflicts
otherwise ack

**Pull request merged**